### PR TITLE
SI-8754  linear seqs aren't

### PR DESCRIPTION
**Pull request by Ichoran**
LinearSeqLike still isn't exactly linear (LinearSeqOptimized is), but at least the docs now give reasonably correct information about what is actually going on.

### CODE REVIEW

File: src/library/scala/collection/LinearSeqOptimized.scala [hunk 1]
```diff
@@ -13,10 +13,22 @@ import mutable.ListBuffer
...
 
-/** A template trait for linear sequences of type `LinearSeq[A]`  which optimizes
+/** A template trait for linear sequences of type `LinearSeq[A]` which optimizes
  *  the implementation of several methods under the assumption of fast linear access.
  *
- *  $linearSeqInfo
+ *  $linearSeqOptim
+ *
+ *  @define  linearSeqOptim
+ *  Linear-optimized sequences are defined in terms of three abstract methods, which are assumed
```
Comment (som-snytt) [inline src/library/scala/collection/LinearSeqOptimized.scala]: `Operations on linear-optimized sequences` might dispel the implication (a nit in the OP) that these are the only abstracts to implement.  Except in so far as every method is an operator.

**Comment by som-snytt** [comment]
LGTM

**Comment by som-snytt** [comment]
It took me a few minutes to get this, although I'm not uninitiated.  The takeaway is that maybe in 2.12, after a deprecation cycle, all linear seqs are optimized; maybe there's an internal linear seq hanging around that is LinearSeqUnoptimized.

I think the litmus test for the doc is whether there is a Coursera exercise using LinearSeq.

Would I be right in guessing there are no implementers out there trying to build a better List?  I mean besides the issue of mutability.

**Comment by Ichoran** [comment]
@som-snytt Improved language again.  I'm not quite sure what it was that it took you a few minutes to get.  This patch certainly _doesn't_ ensure that in 2.12 all linear seqs are actually optimized.  I think it might be true, but I haven't verified.

**Comment by som-snytt** [comment]
The minor confusion was, what makes a linear seq not optimized. One looks at the subclasses in scaladoc to guess.  It LEBTM (even better).

**Pull request merged**